File: csharp/ToolGood.Algorithm/Internals/Functions/MathBase/Function_SQRT.cs
using System;
using System.Text;

namespace ToolGood.Algorithm.Internals.Functions.MathBase
{
	internal class Function_SQRT : Function_1
    {
        public Function_SQRT(FunctionBase func1) : base(func1)
        {
        }

        public override Operand Evaluate(AlgorithmEngine work, Func<AlgorithmEngine, string, Operand> tempParameter)
        {
            var args1 = func1.Evaluate(work, tempParameter); if (args1.IsNotNumber) { args1 = args1.ToNumber("Function '{0}' parameter is error!", "Sqrt"); if (args1.IsError) { return args1; } }
            if (args1.NumberValue < 0) {
                return Operand.Error("Function '{0}' parameter is error!", "Sqrt");
            }
            return Operand.Create(Math.Sqrt(args1.DoubleValue));
        }
        public override void ToString(StringBuilder stringBuilder, bool addBrackets)
        {
            AddFunction(stringBuilder, "Sqrt");
        }
    }

    

}

File: java/src/main/java/toolgood/algorithm/internals/functions/mathbase/Function_SQRT.java
package toolgood.algorithm.internals.functions.mathbase;

import java.lang.StringBuilder;
import toolgood.algorithm.Operand;
import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.internals.functions.FunctionBase;
import toolgood.algorithm.internals.functions.Function_1;

public class Function_SQRT extends Function_1 {
    public Function_SQRT(FunctionBase func1) {
        super(func1);
    }

    @Override
    public Operand Evaluate(AlgorithmEngine work, java.util.function.BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
        Operand args1 = func1.Evaluate(work, tempParameter);
        if (args1.IsNotNumber()) {
            args1 = args1.ToNumber("Function '{0}' parameter is error!", "Sqrt");
            if (args1.IsError()) {
                return args1;
            }
        }
        if (args1.DoubleValue() < 0) {
            return Operand.Error("Function '{0}' parameter is error!", "Sqrt");
        }
        return Operand.Create(Math.sqrt(args1.DoubleValue()));
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        AddFunction(stringBuilder, "Sqrt");
    }
}
